src/Components/Web.JS/src/InputFile.ts

```diff
@@ -105,7 +105,7 @@ export function getFileById(elem: InputElement, fileId: number): BrowserFile {
   const file = elem._blazorFilesById[fileId];
 
   if (!file) {
-    throw new Error(`There is no file with ID ${fileId}. The file list may have changed.`);
+    throw new Error(`There is no file with ID ${fileId}. The file list may have changed. Each selection of file(s) via the InputFile component replaces the prior file(s) selection, selections are not cumulative. For more information please see https://docs.microsoft.com/aspnet/core/blazor/file-uploads.`);
```

Comment:
Note, the URL will be replaced with an aka.ms link for the section containing additional details: https://github.com/dotnet/AspNetCore.Docs/issues/25563

Reply:
```suggestion
    throw new Error(`There is no file with ID ${fileId}. The file list may have changed. Each selection of file(s) via the InputFile component replaces the prior file(s) selected, selections are not cumulative. For more information please see https://aka.ms/blazor-input-file-multiple-selections.`);
```

Reply:
The above is OK with me, though I'd consider making it shorter, so the emphasis is on the docs. We can always update the wording in the docs and can make it as long and detailed as we like there.

I didn't realise before that we already had `The file list may have changed` in this message. Given that's there, I have less sympathy for people who think there's a bug, since we have just told them directly what the problem is. Adding a doc link is pure extra kindness :)

```suggestion
    throw new Error(`There is no file with ID ${fileId}. The file list may have changed. See https://aka.ms/blazor-input-file-multiple-selections.`);
```